src/Component/Form.js

```diff
@@ -0,0 +1,115 @@
+import React, { useState } from 'react';
+import { Grid } from '@material-ui/core';
+import FormControl from '@mui/material/FormControl';
+import Paper from '@mui/material/Paper';
+import Box from '@mui/material/Box';
+import TextField from '@material-ui/core/TextField';
+import AdapterDateFns from '@mui/lab/AdapterDateFns';
+import LocalizationProvider from '@mui/lab/LocalizationProvider';
+import DatePicker from '@mui/lab/DatePicker';
+import Button from '@material-ui/core/Button';
+
+const axios = require('axios');
+const FormData = require('form-data');
+
+function Form({ handleClose }) {
+  const [startDate, setStartDate] = useState('');
+  const [endDate, setEndDate] = useState('');
+  const [reason, setReason] = useState('');
+
+  function dateFormat(date) {
+    let dateF = '';
+    const year = date.getFullYear();
+    const month = date.getMonth() + 1;
+    const day = date.getUTCDate();
+    dateF = `${year}-${month}-${day}`;
+    return dateF;
+  }
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    let Token = localStorage.getItem('django_Token');
+    Token = `token ${Token}`;
+    const data = new FormData();
+    data.append('start_date', dateFormat(startDate));
+    data.append('end_date', dateFormat(endDate));
+    data.append('reason', reason);
+    data.append('user', localStorage.getItem('user_id'));
+
+    const config = {
+      method: 'post',
+      url: 'http://127.0.0.1:8000/leaves/leaves/',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+        Authorization: Token,
+      },
+      data,
+    };
+
+    axios(config)
+      .then((response) => {
+        alert(response.data);
+      })
+      .catch();
+
+    handleClose();
+  };
```

Comment:
Should this be memoized?

Reply:
use useMemo? but on which variable?
without any dependencies give error.

Reply:
Can we use another hook for memoizing this method?